packages/bitcore-node/src/models/transaction.ts

```diff
@@ -274,49 +282,179 @@ export class TransactionModel extends BaseTransaction<IBtcTransaction> {
+          const totalSupplyDecimal = countDecimals(+totalSupply);
+          totalSupply = (
+            BigInt(new Decimal(totalSupply).mul(new Decimal(10 ** totalSupplyDecimal)).toString()) *
+            BigInt(10 ** (decimals - totalSupplyDecimal))
+          ).toString();
+          const token: IToken = {
+            chain,
+            network,
+            txid,
+            contractAddress,
+            decimals,
+            name,
+            symbol,
+            totalSupply,
+          };
+          TokenStorage.collection.updateOne({ txid }, { $set: token }, { upsert: true });
+          result[0].name = name;
+          result[0].decimals = decimals;
+          result[0].symbol = symbol;
+          result[0].totalSupply = totalSupply;
         }
         const contract: IContract = {
           chain,
           network,
           txid,
           contractAddress,
           from: result[0].from,
-          decimals,
-          name,
-          symbol,
-          totalSupply
         };
         ContractStorage.collection.updateOne({ txid }, { $set: contract }, { upsert: true });
-        result[0].name = name;
-        result[0].decimals = decimals;
-        result[0].symbol = symbol;
-        result[0].totalSupply = totalSupply;
+      }
+      if (
+        result.length > 0 &&
+        result[0].log.length > 0 &&
+        result[0].log[0].topics.length > 0 &&
+        result[0].log[0].topics[0] === 'ddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef'
+      ) {
+        result[0].events = [];
+        const receipt = result[0];
+        const from = receipt.log[0].topics[1].replace('000000000000000000000000', '');
+        const to = receipt.log[0].topics[2].replace('000000000000000000000000', '');
+        const value = parseInt(receipt.log[0].data, 16);
+        const contractAddress = receipt.log[0].address;
+        const token = await TokenStorage.collection.findOne({ chain, network, contractAddress });
+        const fromAddress = await fromHexAddress({ address: from, chain, network });
+        const toAddress = await fromHexAddress({ address: to, chain, network });
+        const balanceFrom = await rpc.call('frc20balanceof', [contractAddress, fromAddress]);
+        const balanceTo = await rpc.call('frc20balanceof', [contractAddress, toAddress]);
+        const decimals = token ? token.decimals : await rpc.call('frc20decimals', [contractAddress]);
+        const balanceFromDecimal = countDecimals(+balanceFrom);
+        const balanceToDecimal = countDecimals(+balanceTo);
+        TokenBalanceStorage.collection.updateOne(
+          { contractAddress, address: from },
+          {
+            $set: {
+              chain,
+              network,
+              contractAddress,
+              address: from,
+              balance: (
+                BigInt(new Decimal(balanceFrom).mul(new Decimal(10 ** balanceFromDecimal)).toString()) *
+                BigInt(10 ** (decimals - balanceFromDecimal))
```

Comment:
it could exceed 53 bits, the safe integer

Reply:
ok kub

Reply:
ok kub